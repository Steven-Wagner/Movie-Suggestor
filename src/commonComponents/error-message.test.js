import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import ErrorMessage from './error-message';

describe('<ErrorMessage/>', () => {
    it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
        <Router>
            <ErrorMessage error={{message: 'This is a test Error', button: 'button is rendered'}} />
        </Router>,
        div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(
            <Router>
                <ErrorMessage error={{message: 'This is a test Error', button: 'button is rendered'}}/>
            </Router>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})