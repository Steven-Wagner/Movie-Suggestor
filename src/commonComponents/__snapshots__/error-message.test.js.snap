// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage/> renders the UI as expected 1`] = `
<div
  className="error-message"
>
  This is a test Error
  <button
    className="remote-button title-suggestion"
    onClick={[Function]}
  >
    button is rendered
  </button>
</div>
`;
